app/default/data.py

```diff
@@ -56,6 +56,39 @@ def get_data(endpoint: str, params: dict = None):
     return data
 
 
+def get_data_or_fail_gracefully(endpoint: str, params: dict = None):
+    """
+        Queries the api endpoint provided and returns a
+        data response in json format. Does not return a
+        500 on failure but a 404.
+
+    Args:
+        endpoint (str): an API get data address
+
+    Returns:
+        data (json): data response in json format
+    """
+
+    query_string = ""
+    if params:
+        params = {k: v for k, v in params.items() if v is not None}
+        query_string = urlencode(params)
+        endpoint = endpoint + "?" + query_string
+
+    if Config.USE_LOCAL_DATA:
+        current_app.logger.info(f"Fetching local data from '{endpoint}'.")
+        data = get_local_data(endpoint)
+    else:
+        current_app.logger.info(f"Fetching data from '{endpoint}'.")
+        data = get_remote_data(endpoint)
+    if data is None:
```

Comment:
I wonder if we might be losing some valuable insight here, if the API fails it might be useful to read and record the error returned

Reply:
added logging of errors from service being called 👍🏾